tests/test_monitor.py

```diff
@@ -3,94 +3,73 @@
-
-    # Stub out the real Container class with a mock we can assert on
-    with patch("corgi.monitor.consumer.Container") as mock_container_constructor:
-        # Stub out a different class that we don't want to test here
-        with patch.object(BrewUMBListener, "handler_class") as mock_receiver_constructor:
-            listener.consume()
-
-    # We call the BrewUMBReceiverHandler() constructor with the class's virtual topic addresses
-    # and a dict of selectors for each address
-    mock_receiver_constructor.assert_called_once_with(
-        virtual_topic_addresses=listener.virtual_topic_addresses, selectors=listener.selectors
-    )
-
-    # We call the Container() constructor with an instance of BrewUMBReceiverHandler()
-    # returned by mock_receiver_constructor above so we don't need real UMB certs in tests
-    mock_receiver_instance = mock_receiver_constructor.return_value
-    mock_container_constructor.assert_called_once_with(mock_receiver_instance)
+ADDRESS_PREFIX = f"Consumer.{settings.UMB_CONSUMER}."
 
-    # We call run() on the Container() instance returned by mock_container_constructor
-    mock_container_instance = mock_container_constructor.return_value
-    mock_container_instance.run.assert_called_once_with()
 
-
-def test_brew_umb_receiver_setup():
-    """Test that the BrewUMBReceiverHandler class is set up correctly"""
-    listener = BrewUMBListener()
-    addresses = listener.virtual_topic_addresses
-    selectors = listener.selectors
+def test_dispatcher_setup():
+    """Test UMBDispatcher's configuration"""
     # Stub out the SSLDomain config class to avoid needing real UMB certs in tests
     with patch("corgi.monitor.consumer.SSLDomain") as mock_ssl_domain_constructor:
-        handler = BrewUMBReceiverHandler(virtual_topic_addresses=addresses, selectors=selectors)
+        dispatcher = UMBDispatcher()
 
     # We listen (in client mode) to the address passed in, and connect to the UMB URL from settings
     mock_ssl_domain_constructor.assert_called_once_with(mock_ssl_domain_constructor.MODE_CLIENT)
-    mock_ssl_domain_instance = mock_ssl_domain_constructor.return_value
-    assert handler.virtual_topic_addresses == addresses
-    assert handler.selectors == selectors
-    assert handler.urls == [settings.UMB_BROKER_URL]
+    assert dispatcher.urls == [settings.UMB_BROKER_URL]
 
     # Messages should be accepted manually (only if no exception)
-    assert handler.auto_accept is False
+    assert dispatcher.auto_accept is False
     # Accepted messages should be automatically settled (this is the default)
-    assert handler.auto_settle is True
+    assert dispatcher.auto_settle is True
+
+
+def test_dispatcher_requires_addresses():
+    """Test UMBHandler raises ValueError for missing addresses"""
+
+    class NoAddressHandler(UMBHandler):
+        def __init__(self):
+            with pytest.raises(ValueError):
+                super(UMBHandler, self).__init__(None, None)
```

Comment:
This test makes sense, but I think the implementation won't work - the NoAddressHandler class is only defined and never used by the test. So we'll never call its \_\_init\_\_ method and check the with pytest.raises statement.

Can you just call UMBHandler directly with those None args instead?
```suggestion
def test_dispatcher_requires_addresses():
    """Test UMBHandler raises ValueError for missing addresses"""

    with pytest.raises(ValueError):
        UMBHandler(None, None)
```

Reply:
You're right, I forgot to instantiate the class. :expressionless:  I could call the UMBHandler constructor directly, but I'd prefer to test it with a subclass since that's the intended use.